internal/reader/home_chain.go

```diff
@@ -111,19 +111,38 @@ func (r *homeChainPoller) poll() {
 }
 
 func (r *homeChainPoller) fetchAndSetConfigs(ctx context.Context) error {
-	var chainConfigInfos []ChainConfigInfo
-	err := r.homeChainReader.GetLatestValue(
-		ctx, "CCIPConfig", "getAllChainConfigs", nil, &chainConfigInfos,
-	)
-	if err != nil {
-		return err
+	var allChainConfigInfos []ChainConfigInfo
+	pageIndex := uint64(0)
+	pageSize := uint64(500)
+
+	for {
+		var chainConfigInfos []ChainConfigInfo
+		err := r.homeChainReader.GetLatestValue(
+			ctx,
+			"CCIPConfig",
+			"getAllChainConfigs",
+			map[string]interface{}{
+				"pageIndex": pageIndex,
+				"pageSize":  pageSize,
+			},
+			&chainConfigInfos,
+		)
+		if err != nil {
+			return err
+		}
+		if len(chainConfigInfos) == 0 {
+			break
+		}
+		allChainConfigInfos = append(allChainConfigInfos, chainConfigInfos...)
+		pageIndex++
 	}
-	if len(chainConfigInfos) == 0 {
+
+	if len(allChainConfigInfos) == 0 {
 		// That's a legitimate case if there are no chain configs on chain yet
 		r.lggr.Warnw("no on chain configs found")
 		return nil
 	}
-	r.setState(convertOnChainConfigToHomeChainConfig(chainConfigInfos))
+	r.setState(convertOnChainConfigToHomeChainConfig(allChainConfigInfos))
```

Comment:
q: why are we skipping state sets when there are no configs? Isn't an empty array a valid state?

Reply:
this was an existing approach, yeah in case if the contract function returns empty array
we can still set it but isnt it redundant execution where we can retain initial state as-is
unless there are entries earluer and for some reason the config is removed and state is stale
is this a valid scenario?

Reply:
It could save a null pointer reference though down the line - my thinking is that would be less error-prone

Reply:
updated to set in case if results are empty or non-empty